modules/data/src/main/java/io/fluo/webindex/data/LoadUrls.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.spark.SparkConf;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.archive.io.ArchiveReader;
+import org.archive.io.ArchiveRecord;
+import org.archive.io.warc.WARCReaderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoadUrls {
+
+  private static final Logger log = LoggerFactory.getLogger(LoadUrls.class);
+
+  public static String getFilename(String fullPath) {
+    int slashIndex = fullPath.lastIndexOf("/");
+    if (slashIndex == -1) {
+      return fullPath;
+    }
+    return fullPath.substring(slashIndex + 1);
+  }
+
+  public static void main(String[] args) throws Exception {
+
+    if (args.length != 2) {
+      log.error("Usage: LoadUrls <pathsFile> <range>");
+      System.exit(1);
+    }
+    final String hadoopConfDir = IndexEnv.getHadoopConfDir();
+    final String ccPaths = args[0];
+    if (!(new File(ccPaths).exists())) {
+      log.error("CC paths file {} does not exist", ccPaths);
+      System.exit(1);
+    }
+    int start = 0;
+    int end = 0;
+    try {
+      start = Integer.parseInt(args[1].split("-")[0]);
+      end = Integer.parseInt(args[1].split("-")[1]);
+    } catch (NumberFormatException e) {
+      log.error("Invalid range: {}", args[1]);
+      System.exit(1);
+    }
+    DataConfig dataConfig = DataConfig.load();
+
+    SparkConf sparkConf = new SparkConf().setAppName("CC-LoadUrls");
+    JavaSparkContext ctx = new JavaSparkContext(sparkConf);
+
+    JavaRDD<String> allFiles = ctx.textFile("file://" + ccPaths);
+
+    List<String> copyList = allFiles.takeOrdered(end + 1).subList(start, end + 1);
+
+    log.info("Loading {} files (Range {} of paths file {}) from AWS", copyList.size(), args[1],
+        args[0]);
+
+    JavaRDD<String> copyRDD = ctx.parallelize(copyList, dataConfig.sparkExecutorInstances);
+
+    final String prefix = DataConfig.CC_URL_PREFIX;
+
+    copyRDD.foreachPartition(iter -> {
+      final FluoConfiguration fluoConfig = new FluoConfiguration(new File("fluo.properties"));
+      try (FluoClient client = FluoFactory.newClient(fluoConfig)) {
+        try (LoaderExecutor le = client.newLoaderExecutor()) {
+          iter.forEachRemaining(path -> {
+            String urlToCopy = prefix + path;
+            log.info("Loading {} to Fluo", urlToCopy);
+            try {
+              ArchiveReader reader = WARCReaderFactory.get(new URL(urlToCopy), 0);
+              for (ArchiveRecord record : reader) {
+                Page page = ArchiveUtil.buildPageIgnoreErrors(record);
+                if (page.getOutboundLinks().size() > 0) {
+                  log.info("Loading page {} with {} links", page.getUrl(), page.getOutboundLinks()
+                      .size());
+                  le.execute(PageLoader.updatePage(page));
+                }
+              }
+            } catch (Exception e) {
+              log.error("Exception while processing {}", path, e);
```

Comment:
they will get printed out in the executor logs.. you can get to them via the Spark UI.